### PR TITLE
Initial migration of org plugin to new frontend system 

### PR DESCRIPTION
**Pull request by tudi2d**
Adding basic support for the Org plugin to the new frontend system.

Screenshot 2024-01-31 at 15 32 07

**Example config:**
```yaml
app:
  routes:
    bindings:
      org.catalogIndex: catalog.catalogIndex

  extensions:
    - entity-card:org/group-profile:
        config:
          filter: kind:group
    - entity-card:org/members-list
    - entity-card:org/ownership
    - entity-card:org/user-profile
```

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/org/api-report-alpha.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const _default: BackstagePlugin<
+  {},
+  {
+    catalogIndex: ExternalRouteRef<undefined, false>;
+  }
+>;
+export default _default;
+
+// @alpha (undocumented)
+export const EntityGroupProfileCard: ExtensionDefinition<{
```
Comment (freben) [inline plugins/org/api-report-alpha.md:20]: No need to export the cards actually. Did you forget to run the api reports btw?

File: plugins/org/src/alpha.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  convertLegacyRouteRefs,
+} from '@backstage/core-compat-api';
+import { createPlugin } from '@backstage/frontend-plugin-api';
+import React from 'react';
+import { catalogIndexRouteRef } from './routes';
+import { createEntityCardExtension } from '@backstage/plugin-catalog-react/alpha';
+
+/** @alpha */
+export const EntityGroupProfileCard = createEntityCardExtension({
+  name: 'group-profile',
```
Comment (freben) [inline plugins/org/src/alpha.tsx:28]: remember to add a default filter to all cards where it's applicable

File: plugins/org/src/alpha.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+export default createPlugin({
+  id: 'org',
+  extensions: [
+    EntityGroupProfileCard,
+    EntityMembersListCard,
+    EntityOwnershipCard,
+    EntityUserProfileCard,
+  ],
+  externalRoutes: convertLegacyRouteRefs({
+    catalogIndex: catalogIndexRouteRef,
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply plugins/org/src/alpha.tsx:72]: Nah it'd break the usage in code, unless the code tries to use both the old and new ref

**Review by Rugvip — APPROVED** [review]
Let's :shipit: 🎉

**Pull request merged**